wknml/nml_utils.py

```diff
@@ -57,7 +47,64 @@ def detect_max_node_id_from_all_graphs(graph_dict: Dict[str, nx.Graph]) -> int:
     return max_id
 
 
-def ensure_max_edge_length_in_tree(graph: nx.Graph, max_length: int, current_id: int) -> int:
+def approximate_minimal_edge_length(nml_or_graph: Union[NML, nx.Graph], max_length: int, max_angle: float) -> Union[NML, nx.Graph]:
+    # it is easier to operate on a graph
+    if isinstance(nml_or_graph, NML):
+        nml_graph, parameter_dict = generate_graph(nml_or_graph)
+    else:
+        nml_graph = nml_or_graph[0]
+        parameter_dict = nml_or_graph[1]
+    for group in nml_graph.values():
+        for graph in group:
+            approximate_minimal_edge_length_for_graph(graph, max_length, max_angle)
+
+    # return the same format as the input
+    if isinstance(nml_or_graph, NML):
+        return generate_nml(nml_graph, parameter_dict, globalize_ids=False)
+    else:
+        return nml_graph, parameter_dict
+
+
+def approximate_minimal_edge_length_for_graph(graph: nx.Graph, max_length: int, max_angle: float):
+    nodes_with_degree_of_two = [node for node in graph.nodes if graph.degree(node) == 2]
+
+    for two_degree_node in nodes_with_degree_of_two:
```

Comment:
Maybe just `node_id` (instead of `two_degree_node`) which makes the difference between id and actual node clearer.